packages/browser/src/plugins/segmentio/normalize.ts

```diff
@@ -125,11 +125,16 @@ export function normalize(
   integrations?: LegacySettings['integrations']
 ): object {
   const user = analytics.user()
-  const query = window.location.search
 
   json.context = json.context ?? json.options ?? {}
   const ctx = json.context
 
+  // This guard should not be neccessary -- why would context not exist here? Ditto ^ --
```

Comment:
@pooyaj might have some insight here? 

Thoughts about a broader refactor: It would be simpler / make sense if the page enrichment plugin was was also responsible for this logic:
```
  if (query && !ctx.campaign) {
    ctx.campaign = utm(query)
  }
```

If the current bugfix is safe, then moving the logic altogether should be equally safe / rational.

Reply:
Yeah, I think this is mostly legacy behavior from AJS classic where the UTM params were injected in the Segment integration rather than in AJS core by accessing `window.location.search` directly. I think this change is great. But moving the UTM logic to the page enrichment plugin has some consequences (i.e., UTM params will be sent as a part of the context to all destinations ). I'm not sure if that would be ok or not tho.

Reply:
Thanks, that makes sense. I guess we've decided that, while we pass the query string, we don't want to pass ctx.campaign to all destinations.   

Do you have any insight into this line:
```
json.context = json.context ?? json.options ?? {}
```
The page enrichment plugin already sets context to empty object if it doesn't exist.